pkg/osutil/dns_darwin.go

```diff
@@ -0,0 +1,25 @@
+package osutil
+
+import "github.com/lima-vm/lima/pkg/sysprof"
+
+func DNSAddresses() ([]string, error) {
```

Comment:
QEMU seems using libresolv to get DNS addresses https://gitlab.freedesktop.org/slirp/libslirp/-/blob/v4.6.1/src/slirp.c#L151

Why isn’t this working for us?

Reply:
Because it then only picks the [first one it finds](https://gitlab.freedesktop.org/slirp/libslirp/-/blob/v4.6.1/src/slirp.c#L159). It forwards DNS to that address instead of adding all the servers to the resolver inside the guest (probably because it isn't clear how it could do the latter).

So if the choosen server is not working, or is an [IPv6 address in a broken IPv6 setup](https://github.com/rancher-sandbox/rancher-desktop/issues/572#issuecomment-911805379), then no names will resolve because there is [no fallback](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=625689#28).